Data/Graph.hs

```diff
@@ -23,53 +23,74 @@
 -- Maintainer  :  libraries@haskell.org
 -- Portability :  portable
 --
--- A version of the graph algorithms described in:
+-- = Finite Graphs and Strongly Connected Components
 --
---   /Structuring Depth-First Search Algorithms in Haskell/,
---   by David King and John Launchbury.
+-- The @'Graph'@ type is an adjacency list representation of a finite, directed
+-- graph with @Int@ vertices.
+--
+-- The @'SCC'@ type represents a strongly-connected component of a graph.
+--
+-- == Implementation
+--
+-- The implementation is based on
+--
+--   * /Structuring Depth-First Search Algorithms in Haskell/,
+--     by David King and John Launchbury.
 --
 -----------------------------------------------------------------------------
 
-module Data.Graph(
-
-        -- * External interface
+module Data.Graph (
 
-        -- At present the only one with a "nice" external interface
-        stronglyConnComp, stronglyConnCompR, SCC(..), flattenSCC, flattenSCCs,
+    -- * Graphs
+      Graph
+    , Table
+    , Bounds
+    , Edge
+    , Vertex
 
-        -- * Graphs
+    -- ** Graph Construction
+    , graphFromEdges
+    , graphFromEdges'
+    , buildG
+    , transposeG
 
-        Graph, Table, Bounds, Edge, Vertex,
+    -- ** Graph Properties
+    , vertices
+    , edges
+    , outdegree
+    , indegree
 
-        -- ** Building graphs
+    -- ** Graph Algorithms
+    , dfs
+    , dff
+    , topSort
+    , components
+    , scc
+    , bcc
+    , reachable
+    , path
 
-        graphFromEdges, graphFromEdges', buildG, transposeG,
-        -- reverseE,
 
-        -- ** Graph properties
+    -- * Strongly Connected Components
+    , SCC(..)
```

Comment:
Types are generally introduced before the operations that involve them.

Reply:
Good point, moved this up into the section with the graph types.